src/clj/game/cards/resources.clj

```diff
@@ -2003,68 +2007,29 @@
-                                         (disable-card state :corp second-card))
-                                       ;; enable cards that have left the server
-                                       (when (and (some #{:content} (:zone first-card))
-                                                  (not (some #{server} (:zone first-card)))
-                                                  (some #{server} (:zone second-card)))
-                                         (enable-card state :corp first-card))
-                                       (when (and (some #{:content} (:zone second-card))
-                                                  (not (some #{server} (:zone second-card)))
-                                                  (some #{server} (:zone first-card)))
-                                         (enable-card state :corp second-card))))}
-          run-end-trigger {:event :run-ends
-                           :duration :end-of-run
-                           :effect (effect (enable-server (first (:server target))))}]
-      {:abilities [{:label "Run a remote server"
-                    :cost [(->c :click 1) (->c :trash-can) (->c :brain 1)]
-                    :prompt "Choose a remote server"
-                    :choices (req (cancellable (filter #(can-run-server? state %) remotes)))
-                    :msg (msg "make a run on " target " during which cards in the root of the attacked server lose all abilities")
-                    :makes-run true
-                    :async true
-                    :effect (effect (register-events card [successful-run-trigger
-                                                           run-end-trigger
-                                                           pre-redirect-trigger
-                                                           ;post-redirect-trigger
-                                                           corp-install-trigger
-                                                           swap-trigger])
-                                    (make-run eid target card)
-                                    (disable-server (second (server->zone state target))))}]})))
+                                  :msg "trash all cards in the server for no cost"}]
+    {:abilities [{:label "Run a remote server"
+                  :cost [(->c :click 1) (->c :trash-can) (->c :brain 1)]
+                  :prompt "Choose a remote server"
+                  :choices (req (cancellable (filter #(can-run-server? state %) remotes)))
+                  :msg (msg "make a run on " target " during which cards in the root of the attacked server lose all abilities")
+                  :makes-run true
+                  :async true
+                  :effect (effect
+                            (register-events card [successful-run-trigger])
+                            (make-run eid target card)
+                            (register-lingering-effect
```

Comment:
these have to be in the other order. we don't remove `:end-of-run` static abilities before initiating new runs, so you can add the lingering-effect before calling it. if there's some ordering nonsense here that needs this, then we gotta change how we write these things, because passing the `eid` to an async function _must_ be in tail position.

Reply:
oh yeah, I think we did that as a hack the first time around. I'll fix it to just apply an effect and not be as hacky